conda/recipes/cudf/recipe.yaml

```diff
@@ -0,0 +1,126 @@
+schema_version: 1
+
+context:
+  version: ${{ env.get("RAPIDS_PACKAGE_VERSION") }}
+  minor_version: ${{ (version | split("."))[:2] | join(".") }}
+  cuda_version: ${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[:2] | join(".") }}
+  cuda_major: '${{ (env.get("RAPIDS_CUDA_VERSION") | split("."))[0] }}'
+  date_string: '${{ env.get("RAPIDS_DATE_STRING") }}'
+  py_version: ${{ env.get("RAPIDS_PY_VERSION") }}
+  py_buildstring: ${{ py_version | version_to_buildstring }}
+  head_rev: ${{ git.head_rev(".")[:8] }}
+
+package:
+  name: cudf
+  version: ${{ version }}
+
+source:
+  path: ../../..
+
+build:
+  string: cuda${{ cuda_major }}_py${{ py_buildstring }}_${{ date_string }}_${{ head_rev }}
+  script:
+    content: |
+      ./build.sh cudf
+    secrets:
+      - AWS_ACCESS_KEY_ID
+      - AWS_SECRET_ACCESS_KEY
+      - AWS_SESSION_TOKEN
+    env:
+      CMAKE_C_COMPILER_LAUNCHER: ${{ env.get("CMAKE_C_COMPILER_LAUNCHER") }}
+      CMAKE_CUDA_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CUDA_COMPILER_LAUNCHER") }}
+      CMAKE_CXX_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CXX_COMPILER_LAUNCHER") }}
+      CMAKE_GENERATOR: ${{ env.get("CMAKE_GENERATOR") }}
+      SCCACHE_BUCKET: ${{ env.get("SCCACHE_BUCKET") }}
+      SCCACHE_IDLE_TIMEOUT: ${{ env.get("SCCACHE_IDLE_TIMEOUT") }}
+      SCCACHE_REGION: ${{ env.get("SCCACHE_REGION") }}
+      SCCACHE_S3_USE_SSL: ${{ env.get("SCCACHE_S3_USE_SSL") }}
+      SCCACHE_S3_NO_CREDENTIALS: ${{ env.get("SCCACHE_S3_NO_CREDENTIALS") }}
+      SCCACHE_S3_KEY_PREFIX: cudf-${{ env.get("RAPIDS_CONDA_ARCH") }}
+
+requirements:
+  build:
+    - cmake ${{ cmake_version }}
+    - ninja
+    - ${{ compiler("c") }}
+    - ${{ compiler("cxx") }}
+    - ${{ compiler("cuda") }}
+    - cuda-version =${{ cuda_version }}
+    - ${{ stdlib("c") }}
+  host:
+    - python =${{ py_version }}
+    - pip
+    - cython >=3.0.3
+    - rapids-build-backend >=0.3.0,<0.4.0.dev0
+    - scikit-build-core >=0.10.0
+    - dlpack >=0.8,<1.0
+    - libcudf =${{ version }}
+    - pylibcudf =${{ version }}
+    - rmm =${{ minor_version }}
+    - if: cuda_major == "11"
+      then:
+        - cudatoolkit
+      else:
+        - cuda-cudart-dev
+        - cuda-nvrtc
+        - if: linux64
+          then:
+            - libcufile-dev
+    - cuda-version =${{ cuda_version }}
+  run:
+    - python
+    - typing_extensions >=4.0.0
+    - pandas >=2.0,<2.2.4dev0
+    - cupy >=12.0.0
+    - numba-cuda >=0.4.0,<0.5.0a0
+    - numba >=0.59.1,<0.62.0a0
+    - numpy >=1.23,<2.1
+    - pyarrow>=14.0.0,<20.0.0a0
+    - libcudf =${{ version }}
+    - pylibcudf =${{ version }}
+    - ${{ pin_compatible("rmm", upper_bound="x.x") }}
+    - fsspec >=0.6.0
+    - if: cuda_major == "11"
+      then:
+        - cudatoolkit
+        - ptxcompiler >=0.7.0
+        - cubinlinker  # CUDA enhanced compatibility.
+        - cuda-python >=11.8.5,<12.0a0
+      else:
+        - cuda-cudart
+        # Needed by Numba for CUDA support
+        - cuda-nvcc-impl
```

Comment:
I opened https://github.com/NVIDIA/numba-cuda/issues/146, maybe xref that.

Reply:
noted down in the rattler tracking issue as a followup